app/modules/caching/caching_model.php

```diff
@@ -0,0 +1,91 @@
+<?php
+class Caching_model extends Model
+{
+    public function __construct($serial = '')
+    {
+        parent::__construct('id', 'caching'); //primary key, tablename
+        $this->rs['id'] = '';
+        $this->rs['serial_number'] = $serial; //$this->rt['serial_number'] = 'VARCHAR(255) UNIQUE';
+        $this->rs['collectiondate'] = ""; // Date when data was written
+        $this->rs['expirationdate'] = ""; // Date when data will expire
+        $this->rs['collectiondateepoch'] = 0; // Date when data will expire
+        $this->rs['requestsfrompeers'] = 0;
+        $this->rs['requestsfromclients'] = 0;
+        $this->rs['bytespurgedyoungerthan1day'] = 0;
+        $this->rs['bytespurgedyoungerthan7days'] = 0;
+        $this->rs['bytespurgedyoungerthan30days'] = 0;
+        $this->rs['bytespurgedtotal'] = 0;
+        $this->rs['bytesfrompeerstoclients'] = 0;
+        $this->rs['bytesfromorigintopeers'] = 0;
+        $this->rs['bytesfromorigintoclients'] = 0;
+        $this->rs['bytesfromcachetopeers'] = 0;
+        $this->rs['bytesfromcachetoclients'] = 0;
+        $this->rs['bytesdropped'] = 0;
+        $this->rs['repliesfrompeerstoclients'] = 0;
+        $this->rs['repliesfromorigintopeers'] = 0;
+        $this->rs['repliesfromorigintoclients'] = 0;
+        $this->rs['repliesfromcachetopeers'] = 0;
+        $this->rs['repliesfromcachetoclients'] = 0;
+        $this->rs['bytesimportedbyxpc'] = 0;
+        $this->rs['bytesimportedbyhttp'] = 0;
+        $this->rs['importsbyxpc'] = 0;
+        $this->rs['importsbyhttp'] = 0;
+
+        // Schema version, increment when creating a db migration
+        $this->schema_version = 0;
+        
+        //indexes to optimize queries
+        $this->idx[] = array('collectiondate');
+        $this->idx[] = array('collectiondateepoch');
+        $this->idx[] = array('bytesfromcachetoclients');
+        $this->idx[] = array('bytesfrompeerstoclients');
+        $this->idx[] = array('bytesfromorigintoclients');
+
+        // Create table if it does not exist
+        $this->create_table();
+    }
+
+    // ------------------------------------------------------------------------
+    /**
+     * Process data sent by postflight
+     *
+     * @param string data
+     *
+     **/
+    public function process($data)
+    {
+        // Delete previous entries
+        $this->deleteWhere('serial_number=?', $this->serial_number);
```

Comment:
The data is roughly 21 lines per caching hit or event. At its largest, it shouldn't be any larger than what a busy OS X Server uploads for the servermetrics module. The largest I've seen is roughly 512Kb.